quicklens/src/main/scala/com/softwaremill/quicklens/package.scala

```diff
@@ -115,12 +115,12 @@ package object quicklens {
     def each(fa: F[K, T])(f: T => T): F[K, T]
   }
 
-  implicit def mapQuicklensFunctor[K, T] = new QuicklensMapAtFunctor[Map, K, T] {
-    override def at(fa: Map[K, T], key: K)(f: T => T) = {
-      fa.updated(key, f(fa(key)))
+  implicit def mapQuicklensFunctor[M[KT, TT] <: Map[KT, TT], K, T] = new QuicklensMapAtFunctor[M, K, T] {
+    override def at(fa: M[K, T], key: K)(f: T => T) = {
+      fa.updated(key, f(fa(key))).asInstanceOf[M[K, T]]
```

Comment:
I think some Map implementations (HashMap) have more efficient implementation, as their performance [is guaranteed to be eC](http://docs.scala-lang.org/overviews/collections/performance-characteristics.html) (effectively constant) for them, which would not be possible with a complete rebuild (that would be L - linear).

(This is possible because with immutables as parts of the previous version can be reused for the new version).